kernel/sockdev.c

```diff
@@ -926,3 +936,106 @@ int myst_sockdev_resolve(
 done:
     return ret;
 }
+
+int myst_sockdev_reresolve(
```

Comment:
Some comments about the purpose of the parameters would be helpful.

Would it possible to reuse the `sockdev`, `sock`, `addr`, and `addrlen` instead of new ones? We can clean up the old ones in this function if the socket is re-resolved.

Reply:
Added comments in the header file entry for this function.